tests/devices/qutrit_mixed/test_qutrit_mixed_sampling.py

```diff
@@ -0,0 +1,743 @@
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+
+        spy = mocker.spy(qml.devices.qutrit_mixed.sampling, "_sample_state_jax")
+
+        rng = np.random.default_rng(123)
+        shots = qml.measurements.Shots(shots)
+
+        measurement = qml.sample(wires=[0, 1])
+        res = measure_with_samples(
+            [measurement],
+            batched_qutrit_pure_state,
+            shots,
+            is_state_batched=True,
+            rng=rng,
+            prng_key=jax.random.PRNGKey(184),
+        )
+
+        spy.assert_called()
+
+        assert isinstance(res, tuple)
+        assert len(res) == shots.num_copies
+
+        for s, r in zip(shots, res):
+            assert isinstance(r, tuple)
+            assert len(r) == 1
+            r = r[0]
+
+            assert r.shape == (3, s, 2)
+            assert res[0][0].dtype == np.int64
+
+            # convert to numpy array because prng_key -> JAX -> ArrayImpl -> angry vanilla numpy below
+            r = [np.array(i) for i in r]
+
+            assert_correct_sampled_batched_two_qutrit_pure_state(r)
+
+    # pylint:disable = too-many-arguments
+    @pytest.mark.parametrize("shots", shots_to_test_nonsamples)
+    @pytest.mark.parametrize(
+        "measurement, expected",
+        [
+            (qml.expval(qml.GellMann(1, 3)), np.array([1 / 2, 1, 0])),
+            (qml.var(qml.GellMann(1, 3)), np.array([1 / 4, 0, 1])),
+        ],
+    )
+    def test_nonsample_measure_shot_vector(self, mocker, shots, measurement, expected):
+        """Test that broadcasting works for the other sample measurements and shot vectors"""
+
+        import jax
+
+        spy = mocker.spy(qml.devices.qutrit_mixed.sampling, "_sample_state_jax")
+
+        rng = np.random.default_rng(123)
+        shots = qml.measurements.Shots(shots)
+
+        state = [
+            get_dm_of_state(np.array([[0, 0, 0, 1, 0, 1, 0, 0, 0]]), 2, 2),
+            get_dm_of_state(np.array([1, 0, 0, 1, 0, 0, 1, 0, 0]), 2, 3),
+            get_dm_of_state(np.array([[1, 1, 0, 0, 0, 0, 1, 1, 0]]), 2, 4),
+        ]
+        state = np.stack(state)
+
+        res = measure_with_samples(
+            [measurement],
+            state,
+            shots,
+            is_state_batched=True,
+            rng=rng,
+            prng_key=jax.random.PRNGKey(184),
+        )
+
+        spy.assert_called()
+
+        assert isinstance(res, tuple)
+        assert len(res) == shots.num_copies
+
+        for r in res:
+            assert isinstance(r, tuple)
+            assert len(r) == 1
+            r = r[0]
+
+            assert r.shape == expected.shape
+            assert np.allclose(r, expected, atol=0.01)
+
+
+@pytest.mark.parametrize(
+    "obs",
+    [
+        qml.Hamiltonian([0.8, 0.5], [qml.GellMann(0, 3), qml.GellMann(0, 1)]),
+        qml.s_prod(0.8, qml.GellMann(0, 3)) + qml.s_prod(0.5, qml.GellMann(0, 1)),
+    ],
+)
+class TestHamiltonianSamples:
```

Comment:
Only did the basic cases here, complex Hamiltonian and large amount of wires grows faster with qutrits and I had limited time to add too much